egs/wenetspeech/ASR/pruned_transducer_stateless2/train.py

```diff
@@ -957,6 +961,34 @@ def remove_short_and_long_utt(c: Cut):
         torch.distributed.barrier()
         cleanup_dist()
 
+def display_and_save_batch(
+    batch: dict,
+    params: AttributeDict,
+) -> None:
+    """Display the batch statistics and save the batch into disk.
+
+    Args:
+      batch:
+        A batch of data. See `lhotse.dataset.K2SpeechRecognitionDataset()`
+        for the content in it.
+      params:
+        Parameters for training. See :func:`get_params`.
+      sp:
+        The BPE model.
+    """
+    from lhotse.utils import uuid4
+
+    filename = f"{params.exp_dir}/batch-{uuid4()}.pt"
+    logging.info(f"Saving batch to {filename}")
+    torch.save(batch, filename)
+
+    supervisions = batch["supervisions"]
+    features = batch["inputs"]
+
+    logging.info(f"features shape: {features.shape}")
+
+    num_tokens = params.vocab_size
```

Comment:
https://github.com/k2-fsa/icefall/blob/a2a79c7992f8ef45e1843d419236c5b0b3eb5ac8/egs/wenetspeech/ASR/pruned_transducer_stateless2/train.py#L573-L575

Please print the number of elements in `y`.

Reply:
> https://github.com/k2-fsa/icefall/blob/a2a79c7992f8ef45e1843d419236c5b0b3eb5ac8/egs/wenetspeech/ASR/pruned_transducer_stateless2/train.py#L573-L575
> 
> Please print the number of elements in `y`.

I see, I have modified num_tokens in display_and_save_batch